.github/workflows/lambda_zip.yaml

```
@@ -0,0 +1,35 @@
+name: Create a zipped lambda func for deployment
+
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+    - 'v*' # push events to matching v*
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    # create the zip file
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Build Lambda Function
+        run: |
+          zip lambda.zip rolling-restart.py
+
+      - name: Get release
+        id: get_release
+        uses: bruceadams/get-release@v1.2.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Upload lambda to assets
+        uses: actions/upload-release-asset@v1.0.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.get_release.outputs.upload_url }}
```

You mentioned that this wasn't working for you. I'm pretty sure that's because you were running on tags, not on releases. Different metadata is available.

```suggestion
          upload_url: ${{ github.event.release.upload_url }}
```